src/eShop.ServiceDefaults/OpenApi.Extensions.cs

```diff
@@ -69,26 +38,50 @@ public static IHostApplicationBuilder AddDefaultOpenApi(
         this IHostApplicationBuilder builder,
         IApiVersioningBuilder? apiVersioning = default)
     {
-        var services = builder.Services;
-        var configuration = builder.Configuration;
-        var openApi = configuration.GetSection("OpenApi");
+        var openApi = builder.Configuration.GetSection("OpenApi");
+        var identitySection = builder.Configuration.GetSection("Identity");
+
+        var scopes = identitySection.Exists()
+            ? identitySection.GetRequiredSection("Scopes").GetChildren().ToDictionary(p => p.Key, p => p.Value)
+            : new Dictionary<string, string?>();
+
 
         if (!openApi.Exists())
         {
             return builder;
         }
 
-        services.AddEndpointsApiExplorer();
-
         if (apiVersioning is not null)
         {
             // the default format will just be ApiVersion.ToString(); for example, 1.0.
             // this will format the version as "'v'major[.minor][-status]"
-            apiVersioning.AddApiExplorer(options => options.GroupNameFormat = "'v'VVV");
-            services.AddTransient<IConfigureOptions<SwaggerGenOptions>, ConfigureSwaggerOptions>();
-            services.AddSwaggerGen(options => options.OperationFilter<OpenApiDefaultValues>());
+            var versioned = apiVersioning.AddApiExplorer(options => options.GroupNameFormat = "'v'VVV");
+            string[] versions = ["v1"];
+            foreach (var description in versions)
+            {
+                builder.Services.AddOpenApi(description, options =>
+                {
+                    options.ApplyApiVersionInfo(openApi.GetRequiredValue("Document:Title"), openApi.GetRequiredValue("Document:Description"));
+                    options.ApplyAuthorizationChecks([.. scopes.Keys]);
+                    options.ApplySecuritySchemeDefinitions();
+                    options.ApplyOperationDeprecatedStatus();
+                });
+            }
         }
 
         return builder;
     }
+
+    internal static T GetService<T>(this IServiceCollection services)
```

Comment:
What's this for?

Reply:
We can delete this. It was from an experiment try to get a nicer integration with API versioning working.